application-diagram-ui/src/main/resources/Diagram/DiagramSheet.xml

```diff
@@ -159,6 +161,39 @@ define('xwiki-utils', ['jquery', 'xwiki-meta'], function($, xm) {
       (queryString ? '?' + queryString : '');
     return attachmentURL;
   };
+  
+  var temporaryUploadAttachment = function (blob, attachment) {
+    var documentReference = xm.documentReference;
+    if (typeof attachment !== 'string') {
+      documentReference = attachment.parent;
+      attachment = attachment.name;
+    }
+    var doc = new XWiki.Document(documentReference);
+    var formData = new FormData();
+    formData.append('upload', blob);
```

Comment:
Better to use `formData.append('upload', blob, attachment);`, so to have the filename in the formData

Reply:
Not sure to understand this one: I specify the filename in the params below, so I don't really need it there...

Reply:
This was from when the form was not correct, since an attachment named blob was randomly uploaded. But it is not the case now, the filename specified in params is enough